### PR TITLE
US63877: bgp nonstop routing (nsr) beaker test.

### PR DESCRIPTION
**Pull request by sushirol**

### CODE REVIEW

File: examples/demo_bgp.pp [hunk 1]
```diff
@@ -70,6 +70,7 @@
     bestpath_med_confed                    => true,
     bestpath_med_missing_as_worst          => true,
     bestpath_med_non_deterministic         => $bestpath_med_non_deterministic,
+    nsr                                    => false,
```
Comment (glennmatthews) [inline examples/demo_bgp.pp]: As discussed in scrum this morning - this needs to be a conditional property since it only applies to XR.

File: lib/puppet/type/cisco_bgp.rb [hunk 2]
```diff
@@ -484,6 +487,12 @@ def insync?(is)
     newvalues(:true, :false, :default)
   end # property bestpath_med_non_deterministic
 
+  newproperty(:nsr) do
+    desc 'Enable/Disable nsr'
```
Comment (glennmatthews) [inline lib/puppet/type/cisco_bgp.rb]: How about:

`desc 'Enable/Disable Non-Stop Routing (NSR).'`

File: tests/beaker_tests/cisco_bgp/bgplib.rb [hunk 3]
```diff
@@ -83,6 +87,7 @@ def self.create_bgp_manifest_present(platform, vrf='default')
         bestpath_med_confed                    => 'default',
         bestpath_med_missing_as_worst          => 'default',
 
+
```
Comment (glennmatthews) [inline tests/beaker_tests/cisco_bgp/bgplib.rb]: please revert unneeded whitespace change.

File: tests/beaker_tests/cisco_bgp/bgplib.rb [hunk 4]
```diff
@@ -133,6 +138,10 @@ def self.create_bgp_manifest_present_non_default(platform)
         timer_bestpath_limit                   => '255',
         timer_bestpath_limit_always            => 'true',
         "
+    else
+      conditional_props <<
+        "# Nonstop Routing (NSR)
+        nsr                                    => 'false',"
```
Comment (glennmatthews) [inline tests/beaker_tests/cisco_bgp/bgplib.rb]: Since NSR defaults to false, this test (of non-default values) should be setting it to `true`.

File: tests/beaker_tests/cisco_bgp/test_bgp_non_defaults.rb [hunk 5]
```diff
@@ -146,6 +148,8 @@
   expected_values_vrf1['suppress_fib_pending']                   = 'false'
   expected_values_vrf1['timer_bestpath_limit']                   = '255'
   expected_values_vrf1['timer_bestpath_limit_always']            = 'true'
+else
+  expected_values_vrf1['nsr']                                    = 'false'
```
Comment (glennmatthews) [inline tests/beaker_tests/cisco_bgp/test_bgp_non_defaults.rb]: Seems a bit unnecessary since below you are specifically deleting this key - why bother setting it if you're always going to delete it?

**Comment by glennmatthews** [comment]
Please also add documentation of this property in README.md.

**Comment by sushirol** [comment]
@glennmatthews I have made necessary changes.

File: README.md [hunk 6]
```diff
@@ -518,6 +518,9 @@ Specify Maximum number of AS numbers allowed in the AS-path attribute. Valid val
 ##### `neighbor_down_fib_accelerate`
 Enable/Disable handle BGP neighbor down event, due to various reasons. Valid values are 'true', 'false', and 'default'. This property is not supported on IOS XR.
 
+##### `Non-Stop Routing (NSR)`
+Enable\Disable Non-Stop Routing. Valid values are 'true', 'false', and 'default'. This property is not supported on Nexus.
```
Comment (glennmatthews) [inline README.md]: forward slash please ("Enable/Disable")

File: README.md [hunk 7]
```diff
@@ -518,6 +518,9 @@ Specify Maximum number of AS numbers allowed in the AS-path attribute. Valid val
 ##### `neighbor_down_fib_accelerate`
 Enable/Disable handle BGP neighbor down event, due to various reasons. Valid values are 'true', 'false', and 'default'. This property is not supported on IOS XR.
 
+##### `Non-Stop Routing (NSR)`
```
Comment (glennmatthews) [inline README.md]: This should be the property name `` `nsr` ``.

File: tests/beaker_tests/cisco_bgp/bgplib.rb [hunk 8]
```diff
@@ -36,29 +36,33 @@ def self.create_bgp_manifest_present(platform, vrf='default')
...
-         flush_routes                           => 'default',
-         graceful_restart_helper                => 'default',
-         isolate                                => 'default',
-         maxas_limit                            => 'default',
-         neighbor_down_fib_accelerate           => 'default',
-         shutdown                               => 'default',
-         suppress_fib_pending                   => 'default',
-         timer_bestpath_limit                   => 'default',
-         timer_bestpath_limit_always            => 'default',
+        "bestpath_med_non_deterministic => 'default',
```
Comment (glennmatthews) [inline tests/beaker_tests/cisco_bgp/bgplib.rb]: Please revert these unnecessary whitespace-only changes.

**Comment by sushirol** [comment]
@glennmatthews I have committed necessary changes.

**Comment by glennmatthews** [comment]
:+1:
